### PR TITLE
fix(gatsby-remark-code-repls): handle scoped packages in dependencies…

### PR DESCRIPTION
**Pull request by krzysztofpniak**
… (#12327)
## Description

It allows dependencies with scope like @material-ui/core in the plugin gatsby-remark-code-repls config. That dependencies are used with codesandbox intgration.

## Related Issues

Fixes: #12327

### CODE REVIEW

File: packages/gatsby-remark-code-repls/src/index.js [hunk 1]
```diff
@@ -134,12 +134,10 @@ module.exports = (
             "package.json": {
               content: {
                 dependencies: dependencies.reduce((map, dependency) => {
-                  if (dependency.includes(`@`)) {
-                    const [name, version] = dependency.split(`@`)
-                    map[name] = version
-                  } else {
-                    map[dependency] = `latest`
-                  }
+                  const [, name, , version] = /^(@?.+?)(@(.+))?$/.exec(
```
Comment (DSchau) [inline packages/gatsby-remark-code-repls/src/index.js]: Do you think you could find a cleaner or more self-documenting solution here?

A bit of a nit, but this is really hard to follow.

Comment (krzysztofpniak) [reply packages/gatsby-remark-code-repls/src/index.js]: What about:
```javascript
if (dependency.includes(`@`)) {
  const splitIndex = dependency.findLastIndex('@')
  const name = dependency.substr(0, splitIndex)
  const version = dependency.substr(splitIndex + 1)
  map[name] = version
} else {
  map[dependency] = `latest`
}
```
If you are fine with this version, do you still want a unit test for it? If so, I think its better to extract it to function to a separate file and make some test for it. Anyways, I think after adding unit test I should also cover most complicated cases, where we have two or more @ signs and at least one of them is not at position 0. That test will force me to complicate the code. But I'm leaving this decision up to you.

Comment (DSchau) [reply packages/gatsby-remark-code-repls/src/index.js]: @krzysztofpniak yeah, I'm more aligned with that version, but that also doesn't really work in its current form.

Given:

```js
const getNameAndVersion = (dependency, map) => {
  if (dependency.includes(`@`)) {
    const splitIndex = dependency.findLastIndex('@')
    const name = dependency.substr(0, splitIndex)
    const version = dependency.substr(splitIndex + 1)
    map[name] = version
  } else {
    map[dependency] = `latest`
  }
  return map
}
```

`getNameAndVersion('react')` `{ react: 'latest' }`
`getNameAndVersion('react@16.0.0')` // dependency.findLastIndex is not a function at getNameAndVersion

I'm definitely in favor of breaking out and adding a few unit tests to assert functionality here.

Comment (krzysztofpniak) [reply packages/gatsby-remark-code-repls/src/index.js]: Sorry for the delay. Now it is ok, I also added unit tests.

Comment (pieh) [reply packages/gatsby-remark-code-repls/src/index.js]: I know it's late - but maybe this would be most self documenting approach - https://www.npmjs.com/package/npm-package-arg ;)

Comment (krzysztofpniak) [reply packages/gatsby-remark-code-repls/src/index.js]: @pieh using external libraries for such a small things increases node_modules size. We live in the days when people create libraries for everything but IMHO this is not the way. Of course if library is widely used in project it makes sense. In this situation this is just one call of one simple function.

Comment (DSchau) [reply packages/gatsby-remark-code-repls/src/index.js]: @krzysztofpniak we are sympathetic to that--but using those small little packages also tends to have a lot of other benefits, e.g.

- We don't need to test the code
- We get any fixes (or new features, updates, etc.) for free just by using the package

So for Node.js code--we really recommend using packages if they are available and can be used.

Comment (DSchau) [reply packages/gatsby-remark-code-repls/src/index.js]: Would you be up for making that change? I'd recommend using that library--great idea @pieh

Comment (krzysztofpniak) [reply packages/gatsby-remark-code-repls/src/index.js]: Of course. Done.

**Comment by wardpeet** [comment]
it looks great! thanks a bunch! Do you have time to create a small test for this one to make sure this one is fixed.

**Comment by krzysztofpniak** [comment]
@wardpeet, how this test should look like? I'm a little bit confused. I created tests 8 days ago, then @pieh mentioned `npm-package-arg` and @DSchau [wrote](https://github.com/gatsbyjs/gatsby/pull/12347#discussion_r267024326), that this way tests are not required. So I removed tests I created before.

**Comment by pieh** [comment]
@krzysztofpniak This is mostly for integration - to verify usage of that package (and not package itself) works, let me look up details, push some minimal test and get this merged finally
